spec/categorical_spec.rb

```diff
@@ -0,0 +1,264 @@
+    end
+    
+    context "with index" do
+      context "as array" do
+        let(:dv) do
+          Daru::Vector.new [:a, 1, :a, 1, :c],
+            type: :category,
+            index: ['a', 'b', 'c', 'd', 'e']
+        end
+        subject { dv }
+        
+        its(:index) { is_expected.to be_a Daru::Index }
+        its(:'index.to_a') { is_expected.to eq ['a', 'b', 'c', 'd', 'e'] }
+      end
+      
+      context "as range" do
+        let(:dv) do
+          Daru::Vector.new [:a, 1, :a, 1, :c],
+            type: :category,
+            index: 'a'..'e'
+        end
+        subject { dv }
+          
+        its(:index) { is_expected.to be_a Daru::Index }
+        its(:'index.to_a') { is_expected.to eq ['a', 'b', 'c', 'd', 'e'] }
+      end
+      
+      context "as index object" do
+        let(:tuples) do
+          [
+            [:one, :tin, :bar],
+            [:one, :pin, :bar],
+            [:two, :pin, :bar],
+            [:two, :tin, :bar],
+            [:thr, :pin, :foo]
+          ]
+        end
+        let(:idx) { Daru::MultiIndex.from_tuples tuples }
+        let(:dv) do
+          Daru::Vector.new [:a, 1, :a, 1, :c],
+            type: :category,
+            index: idx
+        end
+        subject { dv }
+        
+        its(:index) { is_expected.to be_a Daru::MultiIndex }
+        its(:'index.to_a') { is_expected.to eq tuples }
+      end
+    end
+  end
+
+  context "#to_category" do
+    let(:dv) { Daru::Vector.new [:a, 1, :a, 1, :c] }
+    subject { dv.to_category }
+
+    it { is_expected.to be_a Daru::Vector }
+    its(:size) { is_expected.to eq 5 }
+    its(:type) { is_expected.to eq :category }
+    its(:ordered?) { is_expected.to eq false }
+    its(:to_a) { is_expected.to eq [:a, 1, :a, 1, :c] }
+  end
+  
+  context "#categories" do
+    let(:dv) { Daru::Vector.new [:a, 1, :a, 1, :c], type: :category }
+    subject { dv.categories }
+    
+    it { is_expected.to be_a Array }
+    its(:size) { is_expected.to eq 3 }
+    its(:'to_a') { is_expected.to eq [:a, 1, :c] }
+  end
+  
+  context "#base_category" do
+    let(:dv) { Daru::Vector.new [:a, 1, :a, 1, :c], type: :category }
+    subject { dv }
+    before { dv.base_category = 1 }
+    
+    its(:base_category) { is_expected.to eq 1 }
+  end
+  
+  context "#coding_scheme" do
+    let(:dv) { Daru::Vector.new [:a, 1, :a, 1, :c], type: :category }
+    subject { dv }
+    before { dv.coding_scheme = :deviation }
+    
+    its(:coding_scheme) { is_expected.to eq :deviation }
+  end
+  
+  context "#categories=" do
+    let(:dv) { Daru::Vector.new [:a, 1, :a, 1, :c], type: :category }
+    subject { dv }
+    before { dv.categories = [1, 2, 3] }
+    
+    its(:to_a) { is_expected.to eq [1, 2, 1, 2, 3] }
+  end
+  
+  context "#order=" do
+    context "valid reordering" do
+      let(:dv) { Daru::Vector.new [:a, 1, :a, 1, :c], type: :category }
+      subject { dv }
+      before { dv.order = [:c, 1, :a] }
```

Comment:
If I understand you plan correctly, I think that `categories` is better, because the user might want to specify which categories are available (even if some have 0 instances), without making them ordered. Then the user could set `categories` to something and set `ordered: false`. On the other hand it would be counterintuitive to set `order` to something, but set `ordered: false` at the same time.